src/shogun/mathematics/linalg/internal/implementation/Scale.h

```diff
@@ -66,6 +66,20 @@ struct scale
 	static void compute(Matrix A, Matrix B, Matrix C, T alpha, T beta);
 };
 
+/** Specialization of scale for the Native backend */
+template<> template<class Matrix>
```

Comment:
extraneous `template <>`. I know we have it in linalg at many places but these are not really necessary. This is simply a partial specialization. We should aim at removing the existing ones. gcc doesn't complain but with clang we get a warning for this.